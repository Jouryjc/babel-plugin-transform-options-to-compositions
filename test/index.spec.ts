import pluginTester from 'babel-plugin-tester'
import path from 'path'
import plugin from '../dist/esm/index'

pluginTester({
	plugin: plugin,
	// fixtures: path.join(__dirname, '__fixtures__'),
	// endOfLine: 'lf'
	tests: {
		props: {
			only: true,
			title: 'props -> defineProps',
			fixture: path.resolve(__dirname, './__fixtures__/props/code.js'),
			snapshot: true
		}
	}
})

	// {
	// 	data: {
	// 		only: true,
	// 		code: `export default {
	// 			data () {
	// 				return {
	// 					lastName: 'Tom'
	// 				}
	// 			}
	// 		}`,
	// 		title: 'data -> reactive',
	// 		snapshot: true,
	// 	},
	// 	props: {
	// 		only: true,
	// 		title: 'props -> defineProps',
	// 		fixture: path.resolve(__dirname, './fixtures/props/input.js'),
	// 		outputFixture: path.resolve(__dirname, './fixtures/props/output.js'),
	// 	}
	// },
