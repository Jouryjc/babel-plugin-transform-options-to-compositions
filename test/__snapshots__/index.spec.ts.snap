// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin props -> defineProps: props -> defineProps 1`] = `

export default {
	props: {
		foo: {
			type: String,
			default: 'foo'
		},
		bar: Number
	}
}

      ↓ ↓ ↓ ↓ ↓ ↓

const props = defineProps({
  foo: {
    type: String,
    default: 'foo',
  },
  bar: Number,
})


`;
